marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -0,0 +1,314 @@
+    [ testPropertyNamed
+        "filter UtxoEvent for Utxos with address in the TargetAddress"
+        "eventsAtAddressTest"
+        eventsAtAddressTest
+
+    , testPropertyNamed
+        "marconi-utxo event-to-sqlRows property"
+        "eventsToRowsRoundTripTest"
+        eventsToRowsRoundTripTest
+
+    , testPropertyNamed
+        "marconi-utxo storable-query address property"
+        "txAddressToUtxoAddressTest"
+        txAddressToUtxoAddressTest
+
+    , testPropertyNamed
+        "marconi-utxo storage-roundtrip property"
+        "utxoStorageTest"
+        utxoStorageTest
+
+    -- TODO BROKEN
+    -- , testPropertyNamed
+    --     "marconi-utxo insert-query property"
+    --     "utxoInsertAndQueryTest"
+    --     utxoInsertAndQueryTest
+
+    , testPropertyNamed
+        "marconi-utxo rewind property"
+        "rewindTest"
+        rewindTest
+
+    , testPropertyNamed
+        "marconi-utxo query-interval property"
+        "utxoQueryIntervalTest"
+        utxoQueryIntervalTest
+
+    , testPropertyNamed
+          "The points that indexer can be resumed from should return at least the genesis point"
+          "propResumingShouldReturnAtLeastTheGenesisPoint"
+          propResumingShouldReturnAtLeastTheGenesisPoint
+
+    , testPropertyNamed
+          "The points that indexer can be resumed from should return at least non-genesis point when some data was indexed on disk"
+          "propResumingShouldReturnAtLeastOneNonGenesisPointIfStoredOnDisk"
+          propResumingShouldReturnAtLeastOneNonGenesisPointIfStoredOnDisk
+
+    , testPropertyNamed
+          "The points that indexer can be resumed from should return an ordered list of points"
+          "propResumingShouldReturnOrderedListOfPoints"
+          propResumingShouldReturnOrderedListOfPoints
+    ]
+
+
+eventsToRowsRoundTripTest :: Property
+eventsToRowsRoundTripTest  = property $ do
+  events <- forAll genUtxoEvents
+  let f :: C.ChainPoint -> IO (Set C.TxIn)
+      f C.ChainPointAtGenesis = pure  Set.empty
+      f _                     = pure . Utxo.ueInputs $ head events
+      rows = concatMap Utxo.eventsToRows events
+  computedEvent <- liftIO . Utxo.rowsToEvents f $ rows
+  let postGenesisEvents = filter (\e -> C.ChainPointAtGenesis /= Utxo.ueChainPoint e ) events
+  length computedEvent === (length . fmap Utxo.ueChainPoint $ postGenesisEvents)
+  Hedgehog.assert (equivalentLists computedEvent postGenesisEvents)
+
+-- Insert Utxo events in storage, and retreive the events
+--
+utxoStorageTest :: Property
+utxoStorageTest = property $ do
+  events <- forAll genUtxoEvents
+  (storedEvents :: [StorableEvent Utxo.UtxoHandle]) <-
+    (liftIO . Utxo.open ":memory:") (Utxo.Depth 10)
+     >>= liftIO . Storable.insertMany events
+     >>= liftIO . Storable.getEvents
+  Hedgehog.assert (equivalentLists storedEvents events)
+
+-- Insert Utxo events in storage, and retrieve the events by address
+--
+utxoInsertAndQueryTest :: Property
```

Comment:
should we comment out this too for now? I get a warning when building the project
```test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs:133:1: warning: [-Wunused-top-binds]
    Defined but not used: ‘utxoInsertAndQueryTest’
    |
133 | utxoInsertAndQueryTest = property $ do
```

Reply:
Thanks, forgot to comment it for now.

Hydra is also giving a failing test case in `marconi-mamba`. Going to fix it.